core/capabilities/ccip/ccip_integration_tests/ccipreader/ccipreader_test.go

```diff
@@ -479,14 +487,86 @@ func TestCCIPReader_Nonces(t *testing.T) {
 	}
 }
 
+func Test_GetChainFeePriceUpdates(t *testing.T) {
+	ctx := testutils.Context(t)
+	s := testSetup(ctx, t, chainD, chainD, nil, evmconfig.DestReaderConfig, nil, false)
+	feeQuoter := deployFeeQuoterWithPrices(t, s.auth, s.sb)
+	// Bind FeeQuoter directly
+	err := s.extendedCR.Bind(ctx, []types.BoundContract{
+		{
+			Address: feeQuoter.Address().String(),
+			Name:    consts.ContractNameFeeQuoter,
+		},
+	})
+	require.NoError(t, err)
+
+	updates := s.reader.GetChainFeePriceUpdate(ctx, []cciptypes.ChainSelector{chainS1})
+	require.Len(t, updates, 1)
+	require.Equal(t, defaultGasPrice, updates[chainS1].Value.Int)
+}
+
+func deployFeeQuoterWithPrices(t *testing.T, auth *bind.TransactOpts, sb *simulated.Backend) *fee_quoter.FeeQuoter {
+	linkAddress := utils.RandomAddress()
+	wethAddress := utils.RandomAddress()
+	address, _, _, err := fee_quoter.DeployFeeQuoter(
+		auth,
+		sb.Client(),
+		fee_quoter.FeeQuoterStaticConfig{
+			MaxFeeJuelsPerMsg:            big.NewInt(0).Mul(big.NewInt(2e2), big.NewInt(1e18)),
+			LinkToken:                    linkAddress,
+			TokenPriceStalenessThreshold: uint32(24 * 60 * 60),
+		},
+		[]common.Address{auth.From},
+		[]common.Address{wethAddress, linkAddress},
+		[]fee_quoter.FeeQuoterTokenPriceFeedUpdate{},
+		[]fee_quoter.FeeQuoterTokenTransferFeeConfigArgs{},
+		[]fee_quoter.FeeQuoterPremiumMultiplierWeiPerEthArgs{
+			{
+				PremiumMultiplierWeiPerEth: 9e17, // 0.9 ETH
```

Comment:
I think this means 90% and not 0.9ETH cc @0xnogo

Reply:
i.e a 10% discount

Reply:
I think this one is set to pay 10% if you are paying with Link token. I think it needs to have 1e18 unit, so what we have here is correct:

```
    // Calculate number of fee tokens to charge.
    // Total USD fee is in 36 decimals, feeTokenPrice is in 18 decimals USD for 1e18 smallest token denominations.
    // Result of the division is the number of smallest token denominations.
    return ((premiumFee * s_premiumMultiplierWeiPerEth[message.feeToken]) + executionCost + dataAvailabilityCost)
      / feeTokenPrice;
      ```

executionCost + dataAvailabilityCost => e36
premium (e18) * s_premiumMultiplierWeiPerEth[message.feeToken] (e18) => e36